samples/infix/writers.cc

```diff
@@ -161,33 +236,31 @@ namespace
             return Literal << (Int ^ "0");
           },
 
-        // Note how we pattern match explicitly for the Error node
-        In(Expression) *
-            (MathsOp
-             << ((T(Expression)[Expression] << T(Error)) * T(Literal))) >>
+        T(Expression) << MathsOp[Op] >>
           [](Match& _) {
-            return err(_(Expression), "Invalid left hand argument");
+            return err(_(Op), "Invalid maths op");
           },
 
-        In(Expression) *
-            (MathsOp
-             << (T(Literal) * (T(Expression)[Expression] << T(Error)))) >>
+        // --- tuple errors ---
+
+        T(Expression) << T(Tuple)[Tuple] >>
           [](Match& _) {
-            return err(_(Expression), "Invalid right hand argument");
+            return err(_(Tuple), "Couldn't evaluate tuple literal");
           },
 
-        In(Expression) *
-            (MathsOp[Op]
-             << ((T(Expression) << T(Error)) * (T(Expression) << T(Error)))) >>
-          [](Match& _) { return err(_(Op), "No valid arguments"); },
+        T(Expression) << T(Append)[Append] >>
+          [](Match& _) {
+            return err(_(Append), "Invalid or unevaluatable tuple append operands");
+          },
 
-        In(Calculation) *
-            (T(Output)[Output] << (T(String) * (T(Expression) << T(Error)))) >>
-          [](Match& _) { return err(_(Output), "Empty output expression"); },
+        T(Expression) << T(TupleIdx)[TupleIdx] >>
+          [](Match& _) {
+            return err(_(TupleIdx), "Invalid or unevaluatable tuple index");
+          },
 
-        In(Calculation) *
-            (T(Assign)[Assign] << (T(Ident) * (T(Expression) << T(Error)))) >>
-          [](Match& _) { return err(_(Assign), "Empty assign expression"); },
+        // --- catch-all: fuzz testing generates expressions that wouldn't be stuck but somehow are
```

Comment:
That comment isn't intended to mean I don't understand it, perhaps poor phrasing on my part.

I mean: because of how the WF is structured, the fuzzer will just put unevaluated (presumably "stuck" and error-ful) expressions anywhere it wants, in ways that the previous pass could never have reached. You could end up with something that means "the previous pass couldn't compute `1 + 1`", despite that fact that it obviously (and testably) can.

I can improve phrasing here.

Reply:
Done.